src/ui/favoritespanewidget.cc

```diff
@@ -1146,10 +1145,36 @@ bool FavoritesModel::removeWordFullPath( const QString & headword )
   return false;
 }
 
+TreeItem * FavoritesModel::getCurrentSelectedOrActiveFolder()
+{
+  QModelIndexList selectedIdxs = selectedIndexes();
+
+  if ( selectedIdxs.size() == 1 ) {
+    QModelIndex idx = selectedIdxs.first();
+    TreeItem * item = getItem( idx );
+    if ( item && item->type() == TreeItem::Folder ) {
+      return item;
+    }
+  }
+
+  return activeFolderFullPath.empty() ? getItem( QModelIndex() ) : getItemByFullPath( activeFolderFullPath );
+}
+
+QModelIndex FavoritesModel::getCurrentSelectedOrActiveFolderIndex()
+{
+  QModelIndexList selectedIdxs = selectedIndexes();
+
+  if ( selectedIdxs.size() == 1 ) {
+    QModelIndex idx = selectedIdxs.first();
+    return idx;
+  }
+
+  return activeFolderFullPath.empty() ? QModelIndex() : getModelIndexByFullPath( activeFolderFullPath );
+}
+
 bool FavoritesModel::isWordPresentFullPath( const QString & headword )
 {
-  TreeItem * targetFolder =
-    activeFolderFullPath.empty() ? getItem( QModelIndex() ) : getItemByFullPath( activeFolderFullPath );
+  TreeItem * targetFolder = getCurrentSelectedOrActiveFolder();
```

Comment:
The ⭐ icon will be wrong again. Based on this code, it needs to update every click.

Reply:
right,this method does not need to change.